engine/src/test/java/io/camunda/zeebe/engine/processing/signal/SignalIntermediateThrowEventTest.java

```diff
@@ -169,4 +171,47 @@ public void shouldApplyOutputMappings() {
     final var signalRecord = RecordingExporter.signalRecords(SignalIntent.BROADCASTED).getFirst();
     Assertions.assertThat(signalRecord.getValue()).hasSignalName(SIGNAL_NAME_1);
   }
+
+  @Test
+  public void shouldNotSendResponseWhenPartOfBatch() {
+    // given
+    final var process =
+        Bpmn.createExecutableProcess(PROCESS)
+            .startEvent()
+            .intermediateCatchEvent(
+                "start", b -> b.message(m -> m.name("start").zeebeCorrelationKey("=businessKey")))
+            .intermediateThrowEvent("signal", b -> b.signal(SIGNAL_NAME_1))
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final var processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS)
+            .withVariables(Map.of("businessKey", "42"))
+            .create();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementId("start")
+                .limit(1)
+                .exists())
+        .isTrue();
+
+    // when
+    ENGINE.message().withName("start").withCorrelationKey("42").publish();
+
+    assertThat(
+            RecordingExporter.signalRecords(SignalIntent.BROADCASTED)
+                .withSignalName(SIGNAL_NAME_1)
+                .limit(1)
+                .exists())
+        .isTrue();
+
+    // then
+    verify(ENGINE.getCommandResponseWriter(), never()).intent(SignalIntent.BROADCASTED);
```

Comment:
❌ To ensure that this call to verify is safe, we must isolate it from the other test cases. We must do that because the engine is reused between the test cases of this class. You can achieve this by adding `Mockito.clearInvocations(ENGINE.getCommandResponseWriter());` at the start of this test method.

Reply:
A very good point, thank you!